front/app/containers/ProjectsShowPage/timeline/CommonGround/CommonGroundStatements.tsx

```diff
@@ -0,0 +1,284 @@
+
+  ${({ $type }) => {
+    switch ($type) {
+      case 'agree':
+        return `
+          left: 20px;
+          top: 50%;
+          transform: translateY(-50%);
+          background-color: ${colors.green500};
+        `;
+      case 'disagree':
+        return `
+          right: 20px;
+          top: 50%;
+          transform: translateY(-50%);
+          background-color: ${colors.red600};
+        `;
+      case 'unsure':
+        return `
+          top: 20px;
+          left: 50%;
+          transform: translateX(-50%);
+          background-color: ${colors.grey600};
+        `;
+    }
+  }}
+`;
+
+interface Props {
+  phaseId: string;
+}
+
+const CommonGroundStatements = ({ phaseId }: Props) => {
+  const { data: progressData, isLoading: isProgressDataLoading } =
+    useCommonGroundProgress(phaseId);
+  const { mutate: reactToIdea } = useReactToStatement(phaseId);
+  const { formatMessage } = useIntl();
+  const isPhone = useBreakpoint('phone');
+
+  const nextIdeaId = progressData?.data.relationships.next_idea.data?.id;
+  const { data: current, isLoading: isIdeaLoading } = useIdeaById(nextIdeaId);
+
+  const handleSwipe = useCallback(
+    (direction: SwipeDirection) => {
+      if (current) {
+        reactToIdea({ ideaId: current.data.id, mode: direction });
+      }
+    },
+    [current, reactToIdea]
+  );
+
+  const {
+    dragOffset,
+    swipeDirection,
+    isAnimating,
+    isDragging,
+    bind,
+    triggerSwipe,
+  } = useSwipeToReact({ onSwipe: handleSwipe });
+
+  if ((nextIdeaId && isIdeaLoading) || isProgressDataLoading) {
+    return (
+      <Box display="flex" justifyContent="center" p="20px">
+        <Spinner />
+      </Box>
+    );
+  }
+
+  if (!current || !nextIdeaId) {
+    return (
+      <Box mt="8px" width="100%" minHeight="260px">
+        <StatementCard
+          $bgColor="white"
+          $transform="translate3d(0px, 0px, 0)"
+          $isAnimating={false}
+          $isDragging={false}
+          border={`1px solid ${colors.grey400}`}
+          style={{ cursor: 'default' }}
+        >
+          <Box
+            display="flex"
+            p="10px 18px"
+            justifyContent="center"
+            alignItems="center"
+            minHeight="100px"
+          >
+            <Text fontSize="l" color="grey800">
+              {formatMessage(messages.noMoreStatements)}
+            </Text>
+          </Box>
+        </StatementCard>
+      </Box>
+    );
+  }
+
+  const transform = `translate3d(${dragOffset.x}px, ${
+    dragOffset.y
+  }px, 0) rotate(${dragOffset.x * 0.05}deg)`;
+
+  const bgColor = (() => {
+    switch (swipeDirection) {
+      case 'up':
+        return `${colors.green500}15`;
+      case 'down':
+        return `${colors.red600}15`;
+      case 'neutral':
+        return `${colors.grey600}15`;
+      default:
+        return 'white';
+    }
+  })();
+
+  return (
+    <Box mt="8px" position="relative" width="100%" minHeight="260px">
+      {[2, 1].map((i) => (
```

Comment:
I'm not not quite sure what "{[2, 1].map((i) => (" is doing here - do you think we could add a short comment maybe?